client/src/app/shared/components/particle-display/particle-display.component.ts

```diff
@@ -0,0 +1,233 @@
+import { BaseViewComponentDirective } from 'app/site/base/base-view';
+
+@Component({
+    selector: 'os-particle-display',
+    templateUrl: './particle-display.component.html',
+    styleUrls: ['./particle-display.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class ParticleDisplayComponent extends BaseViewComponentDirective {
+    public resizeSubject = new Subject<ElementSize>();
+    private resizeAuditTime = 200;
+
+    private particleContainer: Container;
+
+    public set particleImage(imageUrl: string) {
+        this.setParticleImage(imageUrl);
+    }
+
+    public set particleLevel(level: number) {
+        this.setParticleLevel(level);
+    }
+
+    private noAutomaticParticles = {
+        value: 0
+    };
+
+    private slowBlinkingOpacity = {
+        value: 0.8,
+        animation: {
+            enable: true,
+            speed: 1,
+            sync: false,
+            minimumValue: 0.3
+        },
+        random: {
+            enable: true,
+            minimumValue: 0.8
+        }
+    };
+
+    private imageOptions: IImageShape = {
+        replace_color: false,
+        replaceColor: false,
+        src: '',
+        width: 24,
+        height: 24
+    };
+
+    private customImageShape = {
+        type: 'image',
+        image: this.imageOptions
+    };
+
+    private charShapeHearth = {
+        type: 'char',
+        options: {
+            char: {
+                fill: true,
+                font: 'Verdana',
+                weight: '200',
+                style: '',
+                value: ['❤']
+            }
+        }
+    };
+
+    private slightlyRandomSize: any = {
+        value: 16,
+        random: {
+            enable: true,
+            minimumValue: 10
+        }
+    };
+
+    private moveUpOptions = {
+        enable: true,
+        direction: 'top',
+        speed: 1.0,
+        angle: {
+            offset: 45,
+            value: 90
+        },
+        gravity: {
+            enable: true,
+            maxSpeed: 1.5,
+            acceleration: -3
+        },
+        outModes: {
+            left: 'bounce',
+            right: 'bounce',
+            top: 'destroy'
+        }
+    };
+
+    private slowRandomRotation = {
+        value: 0,
+        enable: true,
+        direction: 'random',
+        animation: {
+            enable: true,
+            speed: 9
+        },
+        random: {
+            enable: true,
+            minimumValue: 0
+        }
+    };
+
+    private randomColor = {
+        value: 'random'
+    };
+
+    private singleBottomEmitter = [
+        {
+            direction: 'top',
+            rate: {
+                quantity: 0,
+                delay: 0.33
+            },
+            position: {
+                x: 50,
+                y: 100
+            },
+            size: {
+                mode: SizeMode.percent,
+                width: 100
+            }
+        }
+    ];
+
+    public particlesOptions = {
```

Comment:
Can all options be put into an own configuration file with just `particlesOptions` being exported?